controllers/chat/deleteMessage.js

```diff
@@ -0,0 +1,32 @@
+const {StreamChat} = require('stream-chat');
+
+const deleteMessage = async (req, res) => {
+  const client = new StreamChat(process.env.API_KEY, process.env.SECRET_KEY);
+  const {messageID} = req.params;
+
+  try {
+    await client.connectUser(
+      {
+        id: req.userId
+      },
+      req.token
+    );
+
+    let destroy = await client.deleteMessage(messageID);
+
+    if (!destroy) {
+      return res.status(500).json({message: 'Error deleting chat'});
+    }
+
+    return res.status(200).json({message: 'Message deleted'});
+  } catch (e) {
+    if (e.code === 4) {
+      return res.status(404).json({message: 'Message not found'});
+    }
+    return res.status(500).json({message: 'Error connecting to chat'});
+  } finally {
+    await client.disconnectUser();
+  }
+};
+
+module.exports = deleteMessage;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there are a few areas that could be improved for better error handling and maintainability.



<!-- This is an auto-generated comment by OSS CodeRabbit -->